### PR TITLE
Added RSA key generation supports

### PR DESCRIPTION
**Pull request by lablnet**
Added support of Generate RSA keypair

@peter279k  what do you think?

### CODE REVIEW

File: Tests/Site/KeyTest.php [hunk 1]
```diff
@@ -27,4 +27,10 @@ public function testGenerateEncode(): void
         $key = Key::generateEncode(16);
         $this->assertEquals(32, mb_strlen($key, '8bit'));
     }
+
+    public function testGenerateRsa()
+    {
+    	$rsa = Key::generateRSAKey();
+    	$this->assertNotEmpty($rsa);
```
Comment (peter279k) [inline Tests/Site/KeyTest.php:34]: To make private/public key generating test meaningful, it should assert the public/private key is valid or verified.

File: src/Site/Key.php [hunk 2]
```diff
@@ -74,5 +74,27 @@ public static function generate(int $length = 32): string
...
+	 * @return array
+	 */
+	public static function generateRSAKey(): array
+	{
+		$rsa = openssl_pkey_new(array(
+			"private_key_bits" => 2048,
+			"private_key_type" => \OPENSSL_KEYTYPE_RSA,
+		));
+		openssl_pkey_export($rsa, $rsaPrivateKey);
+		$details = openssl_pkey_get_details($rsa);
```
Comment (peter279k) [inline src/Site/Key.php:93]: Please check this [post](https://paragonie.com/blog/2018/04/protecting-rsa-based-protocols-against-adaptive-chosen-ciphertext-attacks#rsa-doing-it-right) to concern how to generate RSA key correctly.

**Review by peter279k — CHANGES_REQUESTED** [review]
Some request changes are concerned:

- Coding style fix.
- RSA key generating test enhancement.
- RSA key generating approach enhancement.

**Comment by peter279k** [comment]
Consider this [post](https://paragonie.com/blog/2016/12/everything-you-know-about-public-key-encryption-in-php-is-wrong), and I think we don't only use `RSA`.

It should use Hybrid `RSA`+`AES` encryption or other encryption approaches.

**Comment by lablnet** [comment]
> AES

I thought we can use two approaches here

first generate `RSA` keypair 
Other is generate `RSA+AES` Hybrid combinations, so it will be upon user choose which to use....

what do you think?

File: Tests/Site/KeyTest.php (hunk 1 above)
Comment (lablnet) [reply Tests/Site/KeyTest.php:34]: It's not easy to validate key , am sure phpUnit has now such function to do that

Comment (peter279k) [reply Tests/Site/KeyTest.php:34]: It's possible to validate key inside PHPUnit tests.

**Comment by peter279k** [comment]
IMHO, using the hybrid `RSA + AES` encryption instead.

And don't use only `RSA` encryption approach.

File: Tests/Site/KeyTest.php (hunk 1 above)
Comment (peter279k) [reply Tests/Site/KeyTest.php:34]: If possible, it's fine to do this tests on this PR.

The meaningful tests can be on another PR.

**Comment by lablnet** [comment]
This `PR` is now closed, will create new `PR` regarding this.

**Pull request closed without merging**